pkg/sql/schema_changer_test.go

```diff
@@ -405,10 +405,12 @@ func runSchemaChangeWithOperations(
 	wg.Wait() // for schema change to complete.
 
 	// Verify the number of keys left behind in the table to validate schema
-	// change operations.
-	if err := checkTableKeyCount(ctx, kvDB, keyMultiple, maxValue+numInserts); err != nil {
-		t.Fatal(err)
-	}
+	// change operations. This is wrapped in SucceedsSoon to handle cases where
+	// dropped indexes are expected to be GC'ed immediately after the schema
+	// change completes.
+	testutils.SucceedsSoon(t, func() error {
```

Comment:
Was the PrimaryKeyChangeWithOperations flaky now (the old indexes got cleaned up too fast?)